### PR TITLE
implement the pwnstrip function

### PR DESCRIPTION
**Pull request by junmoxiao**
example:
pwn pwnstrip -p alarm -o a.out1 a.out
above command will make alarm has no effect.

### CODE REVIEW

File: pwnlib/commandline/pwnstrip.py [hunk 1]
```diff
@@ -32,23 +34,23 @@ def main(args):
...
 
     for function in args.patch:
         if function not in elf.symbols:
             log.error("Could not find function %r" % function)
-
-        trap = asm(shellcraft.trap())
         offset = elf.symbols[function]
-
-        elf.write(elf.address + offset, trap)
+        elf.asm(offset, 'ret')
```
Comment (zachriggle) [inline pwnlib/commandline/pwnstrip.py:44]: This doesn't work for e.g. ARM or MIPS.  Use `shellcraft.ret`.

File: pwnlib/commandline/pwnstrip.py [hunk 2]
```diff
@@ -32,23 +34,23 @@ def main(args):
...
 
     result = elf.data
 
-    if args.output.isatty():
```
Comment (zachriggle) [inline pwnlib/commandline/pwnstrip.py:48]: Please put this back <3

File: pwnlib/commandline/pwnstrip.py [hunk 3]
```diff
@@ -32,23 +34,23 @@ def main(args):
...
-        result = enhex(result)
+    if args.output:
```
Comment (zachriggle) [inline pwnlib/commandline/pwnstrip.py:48]: `args.output` defaults to sys.stdout, this is unnecessary

File: pwnlib/commandline/pwnstrip.py [hunk 4]
```diff
@@ -32,23 +34,23 @@ def main(args):
...
+        elf.save(args.output)
+        os.chmod(args.output, 0o755)
```
Comment (zachriggle) [inline pwnlib/commandline/pwnstrip.py:50]: This needs to check if `args.output` is a file (`os.path.isfile`) OR needs to catch any exceptions that might get raised.

File: pwnlib/commandline/pwnstrip.py [hunk 5]
```diff
@@ -1,24 +1,26 @@
...
 p = common.parser_commands.add_parser(
     'pwnstrip',
-    help = 'Strip binaries for CTF usage',
+    help='Strip binaries for CTF usage',
 )
 
 g = p.add_argument_group("actions")
 g.add_argument('-b', '--build-id', help="Strip build ID", action='store_true')
-g.add_argument('-p', '--patch', metavar='FUNCTION', help="Patch function", action='append')
-p.add_argument('-o', '--output', type=file, default=sys.stdout)
```
Comment (zachriggle) [inline pwnlib/commandline/pwnstrip.py:19]: Please restore `default=sys.stdout`.

File: pwnlib/commandline/pwnstrip.py [hunk 6]
```diff
@@ -1,24 +1,26 @@
...
+g.add_argument('-p', '--patch', metavar='FUNCTION',
```
Comment (zachriggle) [inline pwnlib/commandline/pwnstrip.py:18]: Unnecessary whitespace changes

File: pwnlib/commandline/pwnstrip.py [hunk 7]
```diff
@@ -32,23 +34,23 @@ def main(args):
         for offset in pwnlib.libcdb.get_build_id_offsets():
             data = elf.read(elf.address + offset + 0xC, 4)
             if data == 'GNU\x00':
-                elf.write(elf.address + offset + 0x10, read('/dev/urandom', 20))
+                elf.write(elf.address + offset + 0x10,
```
Comment (zachriggle) [inline pwnlib/commandline/pwnstrip.py:37]: Unnecessary whitespace changes

**Review by zachriggle — CHANGES_REQUESTED** [review]

**Comment by heapcrash** [comment]
Closing this out since the PR is two+ years old.  Please rebase your changes on top of the current `dev` branch or create a new PR to get this merged.

**Pull request closed without merging**